infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/PrepareStorage.java

```diff
@@ -38,18 +42,23 @@ public PrepareStorage(
       String workspaceId,
       KubernetesEnvironment brokerEnvironment,
       WorkspaceVolumesStrategy volumesStrategy,
-      StartSynchronizer startSynchronizer) {
+      StartSynchronizer startSynchronizer,
+      @Nullable Tracer tracer) {
     this.workspaceId = workspaceId;
     this.brokerEnvironment = brokerEnvironment;
     this.volumesStrategy = volumesStrategy;
     this.startSynchronizer = startSynchronizer;
+    this.tracer = tracer;
+    this.spanName = SPAN_NAME;
   }
 
   @Override
   public List<ChePlugin> execute() throws InfrastructureException {
+    Span tracingSpan = startTracingPhase();
     volumesStrategy.prepare(
         brokerEnvironment, workspaceId, startSynchronizer.getStartTimeoutMillis());
 
+    finishSpanIfExists(tracingSpan);
```

Comment:
If a case of an exception it could be bypassed. Should we move it to finally section?

Reply:
Fixed in same way as for DeployBroker (catch, finish span, rethrow). I don't know if there's a better way to do this though.